### PR TITLE
Only issue a soft error if we fail to convert to placeholder for 30s

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

File: src/libsync/propagateremotemkdir.cpp [hunk 1]
```diff
@@ -158,11 +158,18 @@ void PropagateRemoteMkdir::success()
...
-                Q_ASSERT(result->get() == Vfs::ConvertToPlaceholderResult::Ok);
-                done(SyncFileItem::Success);
-            } else {
-                done(SyncFileItem::FatalError, result->error());
+                switch (result->get()) {
+                case Vfs::ConvertToPlaceholderResult::Ok:
+                    done(SyncFileItem::Success);
+                    return;
+                case Vfs::ConvertToPlaceholderResult::Locked:
+                    done(SyncFileItem::SoftError, tr("UpdateMetadata Timed out"));
```
Comment (dragotin) [inline src/libsync/propagateremotemkdir.cpp:166]: we could go for a more friendly string, such as "Set file status failed due to file lock." maybe? Is the error displayed together with the file name?

File: src/libsync/propagateremotemkdir.cpp [hunk 2]
```diff
@@ -158,11 +158,18 @@ void PropagateRemoteMkdir::success()
         connect(watcher, &QFutureWatcherBase::finished, this, [watcher, this] {
             const auto *result = watcher->result().get();
             if (*result) {
```
Comment (dragotin) [inline src/libsync/propagateremotemkdir.cpp:160]: Check on `result != nullptr` ?

**Review by dragotin — APPROVED** [review]

File: src/libsync/propagateremotemkdir.cpp (hunk 2 above)
Comment (TheOneRing) [reply src/libsync/propagateremotemkdir.cpp:160]: the connect is only triggered if I return something and I really don't intend to return nulll :D

File: src/libsync/propagateremotemkdir.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/libsync/propagateremotemkdir.cpp:166]: yes and I'll change the description

**Pull request merged**